components/RegisterISCN.vue

```diff
@@ -0,0 +1,472 @@
+  return iscnData.value.downloadableUrls
+    .filter(download => download.fileName && download.url)
+    .map((download) => {
+      if (download.fileName && download.type) {
+        return `${download.url}?name=${download.fileName.replace(/\s+/g, '')}`
+      }
+      return ''
+    })
+    .filter(Boolean)
+})
+const payload = computed(() => ({
+  type: iscnData.value.type,
+  name: iscnData.value.title,
+  description: iscnData.value.description,
+  author: iscnData.value.author.name,
+  authorDescription: iscnData.value.author.description,
+  license: iscnData.value.license,
+  contentFingerprints: iscnData.value.contentFingerprints.map(f => f.url),
+  inLanguage: iscnData.value.language,
+  publisher: iscnData.value.publisher,
+  isbn: iscnData.value.isbn,
+  datePublished: iscnData.value.publicationDate
+    ? new Date(iscnData.value.publicationDate).toISOString()
+    : undefined,
+  url: iscnData.value.bookInfoUrl,
+  tagsString: iscnData.value.tags?.join(', ') || '',
+  sameAs: formattedSameAsList.value,
+  thumbnailUrl: iscnData.value.coverUrl
+}))
+
+const isFormValid = computed(() => {
+  const requiredFields = {
+    title: !!iscnData.value.title,
+    description: !!iscnData.value.description,
+    authorName: !!iscnData.value.author.name,
+    contentUrl: !!iscnData.value.contentFingerprints.some(f => !!f.url)
+  }
+
+  error.value = Object.entries(requiredFields)
+    .find(([_, isValid]) => !isValid)?.[0]
+    ?.toUpperCase() || ''
+
+  return Object.values(requiredFields).every(Boolean)
+})
+
+onMounted(() => {
+  const initialData = initializeFromSessionStorage()
+  if (initialData) {
+    iscnData.value = initialData
+  }
+  calculateISCNFee()
+})
+
+const initializeFromSessionStorage = () => {
+  const data = getUploadFileData()
+  if (!data) { return null }
+
+  const baseData = {
+    type: 'book',
+    title: data.epubMetadata?.title || '',
+    description: stripHtmlTags(data.epubMetadata?.description || ''),
+    isbn: '',
+    publisher: '',
+    publicationDate: '',
+    author: {
+      name: data.epubMetadata?.author || '',
+      description: '',
+      url: ''
+    },
+    license: 'All rights reserved',
+    contentFingerprints: [],
+    downloadableUrls: [],
+    coverUrl: data.epubMetadata?.thumbnailArweaveId
+      ? `ar://${data.epubMetadata.thumbnailArweaveId}`
+      : '',
+    language: formatLanguage(data.epubMetadata?.language || ''),
+    tags: data.epubMetadata?.tags || []
+  }
+
+  baseData.downloadableUrls = data.fileRecords
+    .filter(r => r.fileType === 'epub' || r.fileType === 'pdf')
+    .map(file => ({
+      url: file.arweaveKey ? file.arweaveLink : `ar://${file.arweaveId}`,
+      type: file.fileType,
+      fileName: file.fileName
+    }))
+
+  baseData.contentFingerprints = [
+    ...new Set(
+      data.fileRecords
+        .map(r => (r.fileType === 'epub' || r.fileType === 'pdf' ? (r.arweaveKey ? r.arweaveLink : `ar://${r.arweaveId}`) : `ar://${r.arweaveId}`))
```

Comment:
Too complex for ternary, please write as if else function

Reply:
[🎨 Refactor content fingerprint generation](https://github.com/likecoin/nft-book-press/pull/292/commits/db48249f2f0a9eacf54bd94beb5d9ec361e96888)